drivers/bridge/bridge.go

```diff
@@ -46,6 +42,8 @@ type Configuration struct {
 	EnableICC             bool
 	EnableIPForwarding    bool
 	AllowNonDefaultBridge bool
+	// Driver components
+	PortAllocator *portallocator.PortAllocator
```

Comment:
@mavenugo for 1) I think we need to make the portallocator view to be in sync with the kernel. The way to do that is always allocate the port (that we just allocated in the port allocator) also from the os by opening a dummy socket and binding to that port number (or binding without a port number and getting a port allocated from the kernel and then keeping track of it locally)

With the above change, then portallocator New() becomes tied to a namespace(may be we should pass a namespace key or some thing like that) but trying to make portallocator as a whole to become a singleton is not a good idea